galaxy_ng/app/access_control/access_policy.py

```diff
@@ -146,8 +150,66 @@ def get_access_policy(cls, view):
             }
         )
 
+    def _get_public_repo_pks(self):
+        return ansible_models.AnsibleRepository.objects.filter(private=False)
+
+    def _get_private_repo_pks(self, user, perm):
+        return get_objects_for_user(
+            user,
+            perm,
+            ansible_models.AnsibleRepository.objects.filter(private=True),
+        )
+
+    def get_ansible_repository_qs(self, view, qs, repo_perm):
+        qs = ansible_models.AnsibleRepository.objects.all()
+        if view.request.user.has_perm(repo_perm):
+            return qs
+        else:
+            public_repository_pks = self._get_public_repo_pks()
+            private_repository_pks = self._get_private_repo_pks(view.request.user, repo_perm)
+            return qs.filter(Q(pk__in=public_repository_pks) | Q(pk__in=private_repository_pks))
```

Comment:
I think you can merge the querysets rather than doing an exensive `pk__in`: https://github.com/pulp/pulp_container/blob/a003472be52e456114387d9ce6efe39605c1370c/pulp_container/app/viewsets.py/#L1275-L1276

Reply:
I think this should work for your other querysets too

Reply:
I'm getting an error when doing this in `get_ansible_distribution_qs`.  2 out of 3 ain't bad. 

```AssertionError: Cannot combine queries on two different base models.```